packages/block-library/src/navigation/use-is-collapsed.js

```diff
@@ -25,15 +25,49 @@ function useIsCollapsed( overlayMenu, navRef ) {
 
 	const [ isCollapsed, setIsCollapsed ] = useState( shouldBeCollapsed() );
 
+	// We need a layout effect to respond to changed in isMobileBreakPoint.
 	useLayoutEffect( () => {
-		updateIsCollapsed();
 		function updateIsCollapsed() {
 			setIsCollapsed( shouldBeCollapsed() );
 		}
-		window.addEventListener( 'resize', debounce( updateIsCollapsed, 100 ) );
-		return () => {
-			window.removeEventListener( 'resize', updateIsCollapsed );
-		};
+
+		function setIsCollapsedFalse() {
+			setIsCollapsed( false );
+		}
+
+		// This is wrapped in a function so that we can unbind it later.
+		function debouncedSetIsCollapsedFalse() {
+			return debounce( setIsCollapsedFalse, 50 ); // Has to be less than debouncedUpdateIsCollapsed.
+		}
```

Comment:
At this point it's probably worth moving to a `useCallback` to avoid re-creating a debounced function each time effect is called?